### ISSUE TITLE
[Suggestion]: Cleaner missing rule handling

### ISSUE THREAD

**Issue by kunagpal** [eslint/eslint#9129]
**Tell us about your environment**

* **ESLint Version:** 3.19.1 (The issue is not exactly specific to any ESLint version)
* **Node Version:** 8.3.0
* **npm Version:** 5.0.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "rules": [
    "for-direction": "error", // this results in an error, as "for-direction" is not available in eslint v3.x
    "getter-return": "error",
    "switch-colon-spacing": "error"
  ]
}


```

</details>

**What did you do? Please include the actual source code causing the issue.**

N.A

**What did you expect to happen?**
ESLint should show one warning for missing rules or malformed configuration, and continue linting as usual. Alternatively, there could also be a `strict` mode to bail out if a non-existent rule is found in the configuration.

**What actually happened? Please include the actual, raw output from ESLint.**
```
/path/file1.js
  1:1  error  Definition for rule 'for-direction' was not found         for-direction
  1:1  error  Definition for rule 'getter-return' was not found         getter-return
  1:1  error  Definition for rule 'switch-colon-spacing' was not found  switch-colon-spacing

/path/file2.js
  1:1  error  Definition for rule 'for-direction' was not found         for-direction
  1:1  error  Definition for rule 'getter-return' was not found         getter-return
  1:1  error  Definition for rule 'switch-colon-spacing' was not found  switch-colon-spacing
```

As can be seen from the snippet above, the same error messages are repeated across all files. This could be extracted out into one common warning/error header.

**Comment by not-an-aardvark** [comment]
Thanks for the proposal. I have a few thoughts/questions:

* > ESLint should show one warning for missing rules or malformed configuration

    What's your use case for making these errors less strict? If anything, I would have thought we should be *more* strict about missing rules and make them a fatal error, since they indicate a configuration mistake.

* > This could be extracted out into one common warning/error header.

    I'm not sure this is a good idea. One advantage of treating missing rule messages the same as report messages from rules is that users often use custom formatters designed to display report messages, and those formatters can handle missing rule messages too, without needing to be modified. Along similar lines, API methods like `linter.verify` are designed to just return an array of messages for a file, without any additional metadata. Basically, the problem is that there aren't really any existing mechanisms for "common warnings/errors" -- there are only fatal errors (which halt the entire linting run), and normal report messages from rules.

**Comment by kunagpal** [comment]
The one warning suggestion could also be looked at as a fatal error, since a missing/invalid rule applies equally to all files. I didn't think about the custom formatters angle, so this could be placed behind a flag of some sort to prevent breaking existing functionality.

**Comment by nzakas** [comment]
Unfortunately, it looks like there wasn't enough interest from the team or community to implement this change. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to be implemented after 90 days tend to never be implemented, and as such, we [close those issues](https://eslint.org/docs/maintainer-guide/issues#when-to-close-an-issue). This doesn't mean the idea isn't interesting or useful, just that it's not something the team can commit to.
